mailserver/mailserver.go

```diff
@@ -225,9 +241,16 @@ func (s *WMailServer) exceedsPeerRequests(peer []byte) bool {
 
 // processRequest processes the current request and re-sends all stored messages
 // accomplishing lower and upper limits.
-func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bloom []byte) []*whisper.Envelope {
-	ret := make([]*whisper.Envelope, 0)
-	var err error
+func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bloom []byte) (ret []*whisper.Envelope, err error) {
+	// Recover from possible goleveldb panics
+	defer func() {
+		if r := recover(); r != nil {
+			if errString, ok := r.(string); ok {
```

Comment:
I have a little bit different pattern:
```go
var ok bool
err, ok = r.(error)
if !ok {
  err = fmt.Errorf("recovered from panic in processRequest: %v", r)
}
```

Reply:
I just moved the original function here, but maybe we don't need that type assertion. 
what about this?
```
defer func() {
	if r := recover(); r != nil {
		err = fmt.Errorf("recovered from panic in processRequest: %v", r)
	}
}()
```